lenskit-core/src/main/java/org/lenskit/data/dao/file/CollectionEntitySource.java

```diff
@@ -26,12 +26,14 @@
 import org.lenskit.util.io.ObjectStreams;
 
 import java.io.IOException;
+import java.io.Serializable;
 import java.util.Collection;
 
 /**
  * Entity source backed by a collection.
  */
-class CollectionEntitySource implements EntitySource {
+class CollectionEntitySource implements EntitySource, Serializable {
```

Comment:
The use case I'm facing is that I need an interface which gives me a stream of entities. And also, I can restart the stream. This is for iterative training of models. The reason for this design is that algorithms have the control of iteration. I found EntitySource is a good fit. I feel like Source confuses with DAO. Maybe only DAO is enough because it seems that the "file" subfolder in "dao" means a specific file type of daos.